### PR TITLE
Iframes view technical account

### PR DESCRIPTION
**Pull request by jreynard-code**
feat: Add utilities for SimplePowerBIReportEmbed

Add PowerBIEnumFilter and ScenarioDTO for dynamic filters (abstraction layer)
Add PowerBIReportEmbedSimpleFilter and PowerBIReportEmbedMultipleFilter classes
Add PowerBIUtils file

### CODE REVIEW

File: src/PowerBI/PowerBIUtils.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  }
+  return result;
+};
+
+const constructScenarioDTO = (scenario, scenarioList) => {
+  let result;
+  if (scenario !== undefined) {
+    let masterId;
+    if (scenarioList != undefined) {
+      masterId = getMasterScenarioId(scenario,scenarioList);
```
Comment (csm-thu) [inline src/PowerBI/PowerBIUtils.js:54]: Did you check if we can use the **rootId** property ? We could thus remove the _getMasterScenarioId_ function and improve performance for large scenarios lists

**Review by csm-thu — COMMENTED** [review]
Minor improvements suggestions.

Though I wonder if this PowerBI feature should be in the @cosmotech/core module or in the @cosmotech/azure.

**Comment by jreynard-code** [comment]
> Minor improvements suggestions.
> 
> Though I wonder if this PowerBI feature should be in the @cosmotech/core module or in the @cosmotech/azure.

I'll move this to @cosmotech/azure after the modifications ;)

File: src/PowerBI/PowerBIReportEmbedFilter.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+export class PowerBIReportEmbedSimpleFilter {
+  constructor (table, column, acceptedValue) {
+    if (typeof table !== 'string' || typeof column !== 'string') {
+      throw new Error('table and column should be a string');
+    }
```
[the remark this answers is not in the document]
Comment (jreynard-code) [reply src/PowerBI/PowerBIReportEmbedFilter.js:8]: :+1:

File: src/PowerBI/PowerBIReportEmbedFilter.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    this.operator = 'eq';
+    this.values = [acceptedValue];
+  }
+}
+
+export class PowerBIReportEmbedMultipleFilter {
+  constructor (table, column, acceptedValues) {
+    if (typeof table !== 'string' || typeof column !== 'string') {
+      throw new Error('table and column should be a string');
+    }
```
[the remark this answers is not in the document]
Comment (jreynard-code) [reply src/PowerBI/PowerBIReportEmbedFilter.js:20]: :+1:

File: src/PowerBI/PowerBIUtils.js (hunk 1 above)
Comment (jreynard-code) [reply src/PowerBI/PowerBIUtils.js]: rootId corresponds to masterId after checks.
I change that ;)

**Comment by jreynard-code** [comment]
Close PR: Cause the code moved to webapp-component-azure project

**Pull request closed without merging**